### DISCUSSION TITLE
WASM addons support

### DISCUSSION THREAD

**Discussion by caukub** [pelican/panel#16] [Ideas]
Many languages support compiling to WASM so since output is compiled to just one .wasm file, this would be an easy way to distribute addons written in many languages. WASM files can be executed by WASI runtime like Wasmer. Docker WASM support is currently in Beta (https://docs.docker.com/desktop/wasm/).

**Comment by TekExplorer** [comment]
as far as i understand, plugins are basically just php files

perhaps someone can develop a plugin that exposes the apis people will want to wasm?

might need to be automated somehow

**Comment by TekExplorer** [comment]
Consider https://extism.org/

**Comment by lancepioch** [comment]
Thanks, but while PHP can be run in WASM, WASM cannot be run by PHP!

**Reply by TekExplorer** [reply]
not true! it CAN be! Take a second look at extism for an example!

**Reply by syrusakbary** [reply]
Wasm can for sure run inside of PHP. Wasmer-PHP makes this possible

https://github.com/wasmerio/wasmer-php

**Comment by lancepioch** [comment]
Sorry, we've got a full PHP plugin system now!
